.github/workflows/build-test.yml

```
@@ -10,46 +11,215 @@ defaults:
         run: npm ci
 
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@4305c38b25d97ef35a8ad1f985ccf2d2242004f2 # stable
+        with:
+          toolchain: stable
+          targets: ${{ join(matrix.targets, ',') }}
+
+      - name: Cache cargo registry
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
+        with:
+          path: |
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cargo-
+
+      - name: Cache target directory
+        uses: actions/cache@5a3ec84eff668545956fd18022155c47e93e2684 # v4.2.3
+        with:
+          path: target
+          key: ${{ runner.os }}-target-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-target-
+
       - name: Build
-        run: ${{ matrix.settings.build }}
+        run: ${{ matrix.build }}
+
+      - name: Run tests
+        env: # avoid the problems in: https://github.com/rust-lang/rust/issues/78210
+          RUSTFLAGS: "-C target-feature=-crt-static"
+        run: |
+          cargo test --release || { echo "Unit tests failed" >&2; exit 1; }
+
+          # use the --version arg to ensure that the binary compiles and runs
+          test_stdout="$(cargo run --quiet --release -- --version)"
+          exit_code=$?
+
+          if [[ $exit_code -eq 0 ]]; then
+            echo "Binary execution tests passed"
+            echo "Binary version: $test_stdout"
+          else
+            echo "Binary execution tests failed with exit code $exit_code" >&2
+            echo "stdout of sm-action --version was: '$test_stdout'" >&2
+            exit $exit_code
+          fi
+
+      - name: Test `set_env=false`
+        id: set-env-false
+        uses: ./
+        with:
+          access_token: ${{ secrets.TEST_SM_ACCESS_TOKEN }}
+          base_url: "https://vault.euqa.bitwarden.pw"
+          set_env: false
+          secrets: |
+            e9ce5e83-d5b9-46d8-91e8-b1e54d8a52d8 > SECRET_NO_ENV_ONE
+            1e12e623-d4ce-43d2-a3d3-07918217d280 > SECRET_NO_ENV_TWO
+
+      - name: Ensure that `set_env=false` works correctly
+        run: |
+          echo set_env is false, so secrets should not be in env:
+          if [ -n "$SECRET_NO_ENV_ONE" ] || [ -n "$SECRET_NO_ENV_TWO" ]; then
+            echo "Secrets found in environment variables, but set_env is false" >&2
+            exit 1
+          else
+            echo "No secrets found in environment variables, as expected"
+          fi
+          echo
+
+          echo printing unmasked secrets with spaces between each character from GITHUB_OUTPUT:
+          echo "${{ steps.set-env-false.outputs.SECRET_NO_ENV_ONE }}" | sed 's/./& /g'
+          echo "${{ steps.set-env-false.outputs.SECRET_NO_ENV_TWO }}" | sed 's/./& /g'
+          echo
```

To fix the other linter issue

```suggestion
      - name: Ensure that `set_env=false` works correctly
        env:
          SECRET_NO_ENV_ONE: ${{ steps.set-env-false.outputs.SECRET_NO_ENV_ONE }}
          SECRET_NO_ENV_TWO: ${{ steps.set-env-false.outputs.SECRET_NO_ENV_TWO }}
        run: |
          echo set_env is false, so secrets should not be in env:
          if [ -n "$SECRET_NO_ENV_ONE" ] || [ -n "$SECRET_NO_ENV_TWO" ]; then
            echo "Secrets found in environment variables, but set_env is false" >&2
            exit 1
          else
            echo "No secrets found in environment variables, as expected"
          fi
          echo

          echo printing unmasked secrets with spaces between each character from GITHUB_OUTPUT:
          echo "$SECRET_NO_ENV_ONE" | sed 's/./& /g'
          echo "$SECRET_NO_ENV_TWO" | sed 's/./& /g'
          echo
```